src/reconcile/rec_write.c

```diff
@@ -2088,23 +2086,7 @@ __rec_write_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
         if (__wt_ref_is_root(ref))
             break;
 
-        /*
-         * We're about to discard the WT_REF.addr field, and that can race with the cursor walk code
-         * examining tree WT_REFs for leaf pages. Lock the WT_REF unless we know we have exclusive
-         * access.
-         */
-        previous_state = WT_REF_LOCKED; /* -Wuninitialized */
-        if (!F_ISSET(r, WT_REC_EVICT))
-            for (;; __wt_yield()) {
-                previous_state = ref->state;
-                if (previous_state != WT_REF_LOCKED &&
-                  WT_REF_CAS_STATE(session, ref, previous_state, WT_REF_LOCKED))
-                    break;
-            }
-        ret = __wt_ref_block_free(session, ref);
-        if (!F_ISSET(r, WT_REC_EVICT))
-            WT_REF_SET_STATE(ref, previous_state);
-        WT_RET(ret);
+        WT_RET(__wt_ref_block_free(session, ref));
```

Comment:
Is the `WT_REF_LOCK` not required here? In the function `__wt_ref_block_free ()` we are just copying the address and freeing it.

Reply:
This was part of the change, removing the locking that was recently added around the call to `__wt_ref_block_free()`. The rule is any function that reads `WT_REF.addr` is responsible for ensuring that reconciliation (by either checkpoint or eviction), does not happen concurrently with the read. That means that reconciliation should no longer have to lock the WT_REF.addr field.

Reply:
Thanks for the explanation. Understood.